cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisionerMacroEvaluator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package co.cask.cdap.internal.provision;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.macro.InvalidMacroException;
+import co.cask.cdap.api.macro.MacroEvaluator;
+import co.cask.cdap.api.security.store.SecureStore;
+
+/**
+ * Evaluates the secure() macro function for provisioner properties.
+ */
+public class ProvisionerMacroEvaluator implements MacroEvaluator {
+  public static final String SECURE_FUNCTION = "secure";
```

Comment:
will be good to extract this into `Constants` and use it here and also in `DefaultMacroEvaluator`

Reply:
i see, ok.